Bellerophon/Bellerophon/Sources/BPConfig.swift

```diff
@@ -0,0 +1,53 @@
+//
+//  BellerophonConfig.swift
+//  Bellerophon
+//
+//  Created by Satinder Singh on 11/17/17.
+//
+
+import UIKit
+
+/// Allows customization settings for Bellerophon.
+/// If killSwitchView is nil, no kill switch view will be displayed.
+/// If forceUpdateView is nil, no force update view will be displayed.
+/// BellerophonManagerDelegate contains required and optional methods
+public struct BellerophonConfig {
+
+    /// The root window
+    public var window: UIWindow
+
+    /// View to be shown when the kill switch is turned on. Must be set by the application using Bellerophon
+    public var killSwitchView: UIView?
+
+    /// View to be shown when force updated is required. Must be set by the application using Bellerophon
+    public var forceUpdateView: UIView?
+
+    /// Bellerophon manager delegate
+    public weak var delegate: BellerophonManagerDelegate?
+
+    /// Initalizer for `BellerophonManager`
+    /// - Note: It is possible to supply no views and still handle force update and killswitch by utilzing methods found in
+    ///         `BellerophonManagerDelegate` via `shouldForceUpdate` and / or `shouldKillSwitch`
+    /// - Parameters:
+    ///   - window: Root UIWindow
+    ///   - killSwitchView: UIView to display when kill switch is triggered. If nil, no view will be shown for kill switch event.
+    ///   - forceUpdateView: UIView to display when force update is triggered. If nil, no view will be shown for force update event.
+    ///   - delegate: `BellerophonManagerDelegate` with required and optional callbacks.
+    public init(window: UIWindow,
+                killSwitchView: UIView? = nil,
+                forceUpdateView: UIView? = nil,
+                delegate: BellerophonManagerDelegate) {
```

Comment:
With `delegate` being optional, shouldn't this also be optional and default to nil?

Reply:
There is a required method that triggers the actual network call so the delegate is required